packages/webcert/src/feature/certificate/Inputs/UeCauseOfDeath/UeCauseOfDeath.tsx

```diff
@@ -11,18 +19,22 @@ interface Props {
 }
 
 const UeCauseOfDeath: React.FC<Props> = ({ disabled, question }) => {
-  const dispatch = useDispatch()
+  const dispatch = useAppDispatch()
   const config = question.config as ConfigureUeCauseOfDeath
   const validationErrors = useSelector(getVisibleValidationErrors(question.id))
   const isShowValidationError = useSelector(getShowValidationErrors)
   const [currentValue, setCurrentValue] = useState<ValueCauseOfDeath>(question.value as ValueCauseOfDeath)
 
   const handleChange = (value: ValueCauseOfDeath) => {
     setCurrentValue(value)
+    let debut = value.debut
+    if (value && !isValid(getValidDate(value.debut.date))) {
+      debut = { id: value.id, type: CertificateDataValueType.DATE, date: '' }
+    }
```

Comment:
`id` should be `value.debute.id`, and `getValidDate` manipulates the date string before validating so it needs to be sent as the result.

```typescript
const date = getValidDate(value.debut.date)
const debut = { ...value.debut,  date: isValid(date) ? date : '' }
```

Reply:
Fixed this, but since getValidDate returns Date, it needs to be converted back to string, or we'll get an error